test/org/zaproxy/zap/extension/websocket/client/WebSocketProxyUnitTest.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2018 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.zaproxy.zap.extension.websocket.client;
+
+import org.apache.commons.httpclient.URI;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import static org.junit.Assert.assertTrue;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.websocket.ExtensionWebSocket;
+import org.zaproxy.zap.extension.websocket.WebSocketMessage;
+import org.zaproxy.zap.extension.websocket.WebSocketObserver;
+import org.zaproxy.zap.extension.websocket.WebSocketProxy;
+import org.zaproxy.zap.testutils.WebSocketTestUtils;
+import org.zaproxy.zap.testutils.websocket.server.NanoWebSocketConnection;
+
+public class WebSocketProxyUnitTest extends WebSocketTestUtils {
+	
+	private final static int WEBSOCKET_PORT = 8885;
+	private final static String HOST_NAME = "localhost";
+	
+	@Before
+	public void openWebSocketServer() throws Exception {
+		super.startWebSocketServer(HOST_NAME,WEBSOCKET_PORT);
+		super.setUpZap();
+	}
+	
+	@Override
+	protected void setUpMessages() {
+		mockMessages(new ExtensionWebSocket());
+	}
+	
+	@Test
+	public void shouldResponsePong() throws Exception {
+		ServerConnectionEstablisher establisher = new ServerConnectionEstablisher();
+		HttpMessage handshakeRequest = new HttpMessage(HttpHandshakeBuilder.getHttpHandshakeRequestHeader(new URI("http://" + HOST_NAME + ":" + WEBSOCKET_PORT,true)));
+		
+		WebSocketProxy webSocketProxy = establisher.send(new HandshakeConfig(handshakeRequest,false,false));
+		NanoWebSocketConnection webSocketConnection = super.getLastConnection();
+		
+		webSocketConnection.setPingScheduling(50,("1010").getBytes());
+		
+		webSocketProxy.addObserver(new WebSocketObserver() {
```

Comment:
This should be removed, instead we should always assert that the client received the Pong (check opcode and contents, `webSocketConnection.popIncomingMessage()` should give the latest Pong).